app/models/authentication.rb

```diff
@@ -68,6 +68,26 @@ def self.db_name
     "auth_key_pair_cloud"
   end
 
+  # Builds a case statement that case be used in a sql ORDER BY.
+  #
+  # Generated SQL looks like:
+  #
+  #   CASE
+  #   WHEN LOWER(status) = ''      THEN -1
+  #   WHEN LOWER(status) = 'valid' THEN 0
+  #   ...
+  #   ELSE -1
+  #
+  def self.status_severity_arel
+    arel_case = Arel::Nodes::Case.new
```

Comment:
obscurity contest:
```ruby
STATUS_SEVERITY.each_with_object(Arel::Nodes::Case.new) do |(value, sort_weight), arel_case|
  arel_case.when(arel_table[:status].lower.eq(value)).then(sort_weight)
end.else(-1)
```

Reply:
wonder if that `lower` is really necessary.
Would read much better without it

Reply:
> wonder if that lower is really necessary.

Matching what is done in [`Authentication#status_severity`](https://github.com/ManageIQ/manageiq/blob/421c300b/app/models/authentication.rb#L71-L73).  As to whether it is necessary, I would rather address at another time.